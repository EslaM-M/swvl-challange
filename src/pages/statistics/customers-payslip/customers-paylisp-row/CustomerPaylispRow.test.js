import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CustomerPaylispRow from "./index";
describe("CustomerPaylispRow Component", () => {
  test("Component snapshot render correctly", () => {
    const container = render(
      <table>
        <tbody>
          <CustomerPaylispRow
            index={0}
            PaymentType="credit"
            currency="EGP"
            price="45"
            pickupStation="1"
            userId="244"
            userName="Amr Hassan"
          />
        </tbody>
      </table>
    );
    expect(container).toMatchSnapshot();
  });

  test("Component Render Data Correctly", async () => {
    const { getByTestId } = render(
      <table>
        <tbody>
        <CustomerPaylispRow
            index={0}
            PaymentType="credit"
            currency="EGP"
            price="45"
            pickupStation="1"
            userId="244"
            userName="Amr Hassan"
          />
        </tbody>
      </table>
    );
    expect(getByTestId("index").innerHTML).toBe("1");
    expect(getByTestId("user-id").innerHTML).toBe("244");
    expect(getByTestId("user-name").innerHTML).toBe("Amr Hassan");
    expect(getByTestId("payment-type").innerHTML).toBe("credit");
    expect(getByTestId("price").innerHTML).toBe("45");
    expect(getByTestId("currency").innerHTML).toBe("EGP");
    expect(getByTestId("station").innerHTML).toBe("Station1");
  });
});
