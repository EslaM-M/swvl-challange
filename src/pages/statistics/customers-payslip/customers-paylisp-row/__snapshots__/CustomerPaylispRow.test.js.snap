// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerPaylispRow Component Component snapshot render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr>
            <td
              data-testid="index"
            >
              1
            </td>
            <td
              data-testid="user-id"
            >
              244
            </td>
            <td
              data-testid="user-name"
            >
              Amr Hassan
            </td>
            <td
              data-testid="payment-type"
            >
              credit
            </td>
            <td>
              <label
                data-testid="price"
              >
                45
              </label>
              <label
                data-testid="currency"
              >
                EGP
              </label>
            </td>
            <td
              data-testid="station"
            >
              Station
              1
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr>
          <td
            data-testid="index"
          >
            1
          </td>
          <td
            data-testid="user-id"
          >
            244
          </td>
          <td
            data-testid="user-name"
          >
            Amr Hassan
          </td>
          <td
            data-testid="payment-type"
          >
            credit
          </td>
          <td>
            <label
              data-testid="price"
            >
              45
            </label>
            <label
              data-testid="currency"
            >
              EGP
            </label>
          </td>
          <td
            data-testid="station"
          >
            Station
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
